### PR TITLE
Add helper functions and docstrings to ascii.zig

### PR DESCRIPTION
**Pull request by gpanders**
* Add an upper case variant of `allocLowerString`
  * ~~Add case-sensitive variants of `eqlIgnoreCase`, `indexOfIgnoreCase`,
    and `indexOfIgnoreCasePos`~~
  * Add and update docstrings on functions

### CODE REVIEW

File: lib/std/ascii.zig [hunk 1]
```diff
@@ -255,20 +274,43 @@ test "eqlIgnoreCase" {
     std.testing.expect(!eqlIgnoreCase("hElLo!", "helro!"));
 }
 
-/// Finds `substr` in `container`, starting at `start_index`.
-/// TODO boyer-moore algorithm
-pub fn indexOfIgnoreCasePos(container: []const u8, start_index: usize, substr: []const u8) ?usize {
+/// Return `true` if `a` and `b` are equal, case sensitive.
+pub fn eqlMatchCase(a: []const u8, b: []const u8) bool {
```
Comment (daurnimator) [inline lib/std/ascii.zig:278]: This function should not exist: it's just `mem.eql(u8, a, b)`

File: lib/std/ascii.zig [hunk 2]
```diff
@@ -281,3 +323,22 @@ test "indexOfIgnoreCase" {
 
     std.testing.expect(indexOfIgnoreCase("FOO foo", "fOo").? == 0);
 }
+
+/// Finds `substr` in `container`, case sensitive, starting at `start_index`.
+pub fn indexOfMatchCasePos(container: []const u8, start_index: usize, substr: []const u8) ?usize {
```
Comment (daurnimator) [inline lib/std/ascii.zig:328]: These `MatchCase` functions should not exist: this one is just `mem.indexOfPos`

Comment (gpanders) [reply lib/std/ascii.zig]: You're right -- I'd wondered if there was a reason these did not exist already, and of course that is why.

This PR now just contains the addition of the allocUpperString and some docstring additions/changes. I admit it's a bit underwhelming. If it'd be helpful, I can go through and add docstrings to other functions that are currently missing them and turn this into a "documentation" PR.

File: lib/std/ascii.zig (hunk 1 above)
Comment (gpanders) [reply lib/std/ascii.zig]: Removed in 9f496c077766172465a065eafba56dd5280021a6

**Pull request merged**